target_chains/ethereum/contracts/contracts/entropy/Entropy.sol

```diff
@@ -216,45 +208,80 @@ abstract contract Entropy is IEntropy, EntropyState {
+    // in the arguments. It returns a sequence number.
     //
-    // Note that this function can only be called once per in-flight request. Calling this function deletes the stored
-    // request information (so that the contract doesn't use a linear amount of storage in the number of requests).
-    // If you need to use the returned random number more than once, you are responsible for storing it.
+    // The address calling this function should be a contract that inherits from the IEntropyConsumer interface.
+    // The `entropyCallback` method on that interface will receive a callback with the generated random number.
     //
-    // This function must be called by the same `msg.sender` that originally requested the random number. This check
-    // prevents denial-of-service attacks where another actor front-runs the requester's reveal transaction.
-    function reveal(
+    // This method will revert unless the caller provides a sufficient fee (at least getFee(provider)) as msg.value.
+    // Note that excess value is *not* refunded to the caller.
+    function requestWithCallback(
         address provider,
-        uint64 sequenceNumber,
-        bytes32 userRandomness,
-        bytes32 providerRevelation
-    ) public override returns (bytes32 randomNumber) {
-        EntropyStructs.Request storage req = findRequest(
+        bytes32 userRandomNumber
+    ) public payable override returns (uint64 assignedSequenceNumber) {
+        EntropyStructs.Request storage req = requestHelper(
             provider,
-            sequenceNumber
+            constructUserCommitment(userRandomNumber),
+            // If useBlockHash is set to true, it allows a scenario in which the provider and miner can collude.
+            // If we remove the blockHash from this, the provider would have no choice but to provide its committed
+            // random number. Hence, useBlockHash is set to false.
+            false,
+            true
         );
-        // Check that there is an active request for the given provider / sequence number.
-        if (
-            req.sequenceNumber == 0 ||
-            req.provider != provider ||
-            req.sequenceNumber != sequenceNumber
-        ) revert EntropyErrors.NoSuchRequest();
+        assignedSequenceNumber = req.sequenceNumber;
 
-        if (req.requester != msg.sender) revert EntropyErrors.Unauthorized();
+        emit RequestedWithCallback(
+            provider,
+            req.requester,
+            assignedSequenceNumber,
+            userRandomNumber,
+            req
+        );
+    }
 
+    // This method validates the provided user's revelation and provider's revelation against the corresponding
+    // commitment in the in-flight request. If both values are validated, this method will update the provider
+    // current commitment and returns the generated random number.
+    function revealHelper(
```

Comment:
what this function does is pretty clear. We can rename it something like `revealAndClearRequest`

Reply:
similarly here